jiant/utils/data_loaders.py

```diff
@@ -97,27 +97,36 @@ def load_tsv(
 ):
     """
     Load a tsv.
-    To load only rows that have a certain value for a certain column,
-    like genre in MNLI, set filter_idx and filter_value (for example,
-    for mnli-fiction  we want columns where genre == 'fiction' ).
+
+    To load only rows that have a certain value for a certain columnn, set filter_idx and
+    filter_value (for example, for mnli-fiction we want rows where the genre column has
+    value 'fiction').
+
     Args:
-        s1_idx; int
-        s2_idx (int|None): if not None, look for sentence2 at s2_idx.
-                           else, return empty list
-        targ_idx: int
-        has_labels: if False, don't look for labels at position label_idx.
-                    No value for labels will be returned.
-        filter_idx: int this is the index that we want to filter from
-        filter_value: string the value in which we want filter_idx to be equal to
-        return_indices: bool that describes if you need to return indices
-            (for purposes of matching)
-        label_fn is a function that expects a row and outputs the label
-        tag_vocab is a allenlp vocab object contains the tags
-        tag2idx_dict is a <string, int> dictionary from coarse category name to column index
+        tokenizer_name (str): The name of the tokenizer to use (see defaluts.conf for values).
+        data_file (str): The path to the file to read.
+        max_seq_len (int): The maximum number of tokens to keep after tokenization, per text field.
+            Start and end symbols are introduced before tokenization, and are counted, so we will
+            keep max_seq_len - 2 tokens *of text*.
+        label_idx (int|None): The column index for the label field, if present.
+        s1_idx (int): The column index for the first text field.
+        s2_idx (int|None): The column index for the second text field, if present.
+        label_fn (fn: str -> int|None): A function to map items in column label_idx to int-valued
+            labels.
+        col_indices (list[int]): TODO(YADA): ???
```

Comment:
Right now, we do not red in the column names in pandas so we rely on indices. Tag_vocab I believe has to do with CCG-related tags where we want to evaluate on a subset of the dataset (@HaokunLiu)

Reply:
Ah, sorry, I didn't notice this. There are too many github mails.

Reply:
Tag_vocab is not CCG-related, it's related to cola-analysis etc., where we want to know the results on a subset of the task examples.

Reply:
Tx. If you know what it's for/what it requires, please push a comment to this branch. :D

Reply:
The tag means, in some dataset, each example may have one or more tags, like mnli-diagnostic and cola-analysis. Tag_vocab is a vocab used to track what tags are there all together in different data splits of that task. tag2idx_dict means, the data tsv, some columns contains information about what tags the example have. And for both mnli-diagnostic and cola-analysis, the tags forms a two-level hierarchy, there are "big tags" beneath which there are "smaller tags", this dictionary is a mapping from the name of "big tags" to index of column contain the information about what "small tags" beneath that "big tag" the examples have.
Initially this mechanism was introduced when working on NPI task, as well as for refactoring cola-analysis and mnli-diagnostic, but the mnli-diagnostic refactoring thing get dropped in a consecutive jiant upgrades, superglue upgrades, etc.

Reply:
Would you mind updating the branch with this information? I'm not using this code myself now, I'm asking because there really needs to be documentation in the docstring. Thanks!

Reply:
Ok, you mean update mnli-path-cleanup branch?

Reply:
@pruksmhc Mind updating the TODO on this line?